### PR TITLE
Display html books

### PR DESCRIPTION
**Pull request by chrisvire**
- [x] Convert HTML books
  - [x] Update image tags to reference `/illustrations`
  - [x] Use BUILD_BASE_PATH for path to illustrations
- [x] HTML books display in the Book Selector 
- [x] When an HTML book is selected
  - [x] Display like scripture (with gutters on desktop)
  - [x] Chapter selector hidden
  - [x] Display title in book selector
  - [x] No swiping or next/prev buttons
  - [x] Use styles in `<head>` of the HTML file
  - [x] Display correctly on mobile
  - [x] Font size adjustment

Sample project provided by user: https://www.evangelischebijbelvertaling.nl/EBV.zip

### CODE REVIEW

**Comment by chrisvire** [comment]
I need to fix display on moble!

image

Here it is on desktop:

image

File: src/lib/data/navigation.ts [hunk 1]
```diff
@@ -119,8 +122,17 @@ export class NavigationContext {
 
     private updateHeadings() {
         const document = this.catalog.documents.find((b) => b.bookCode === this.book);
-        this.title = document.toc;
-        this.name = document.h;
+        if (document) {
+            this.title = document.toc;
+            this.name = document.h;
+            return;
```
Comment (davidmoore1) [inline src/lib/data/navigation.ts:128]: Are you still not going to find the html book when searching the catalog?  Not sure what this and the next one are doing

File: src/lib/data/stores/scripture.js [hunk 2]
```diff
@@ -151,11 +151,11 @@ export const fontChoices = derived(refs, ($refs) => {
     if (!$refs.initialized) return [];
     const bookFonts = config.bookCollections
         .find((x) => x.id === $refs.collection)
-        .books.find((x) => x.id === $refs.book).fonts;
+        .books.find((x) => x.id === $refs.book)?.fonts;
     const colFonts = config.bookCollections.find((x) => x.id === $refs.collection).fonts;
     const allFonts = [...new Set(config.fonts.map((x) => x.family))];
     const currentFonts =
-        bookFonts.length > 0 ? bookFonts : colFonts.length > 0 ? colFonts : allFonts;
+        bookFonts?.length > 0 ? bookFonts : colFonts.length > 0 ? colFonts : allFonts;
```
Comment (davidmoore1) [inline src/lib/data/stores/scripture.js:158]: what happens to this comparison if there is no book fonts.  Is it undefined?

File: src/lib/scripts/configUtils.ts [hunk 3]
```diff
@@ -64,6 +64,7 @@ export function getStyle(configData: any, option: string, bc: string, book: stri
     const bookData = bcData.books.find((x) => x.id === book);
     // Use the style of the book, if defined.
     if (
+        bookData &&
```
Comment (davidmoore1) [inline src/lib/scripts/configUtils.ts:67]: I had just hit this one too yesterday.  Looks good.

**Review by davidmoore1 — APPROVED** [review]
Generally, approve with questions

File: convert/convertBooks.ts [hunk 4]
```diff
@@ -386,6 +420,20 @@ export type Quiz = {
...
+    const dstFile = path.join('static', 'collections', context.bcId, book.file);
+
+    let content = fs.readFileSync(srcFile, 'utf-8');
+    content = applyFilters(content, htmlFilterFunctions, context.bcId, book.id);
+    files.push({
+        path: dstFile,
+        content
+    });
+
+    //process.stdout.write(`copyHtmlBook: bookId:${book.id}, src:${srcFile}, dst:${dstFile}\n`);
```
[the remark this answers is not in the document]
Comment (chrisvire) [reply convert/convertBooks.ts:434]: Can be deleted.

File: src/lib/data/navigation.ts (hunk 1 above)
Comment (chrisvire) [reply src/lib/data/navigation.ts:128]: The old code was pulling out the `title` and `name` properties from the document in the catalog. I tried to keep it consistent.  However, I don't see where these are even being used.

**Review by davidmoore1 — APPROVED** [review]

**Pull request merged**